security/pkg/platform/aws_test.go

```diff
@@ -56,39 +59,302 @@ func TestIsProperPlatform(t *testing.T) {
+				"LAogICJyYW1kaXNrSWQiIDogbnVsbCwKICAiaW1hZ2VJZCIgOiAiYW1pLWZhYmY1YzgyIiwKICAicGVuZGluZ1" +
+				"RpbWUiIDogIjIwMTctMDgtMjdUMTc6MTg6MjBaIiwKICAicmVnaW9uIiA6ICJ1cy13ZXN0LTIiCn0=\"",
+		},
+	}
 
-	c := ec2metadata.New(unit.Session, &aws.Config{Endpoint: aws.String(server.URL + "/latest")})
-	na := &AwsClientImpl{client: c}
+	for id, c := range testCases {
+		sigBytes, err := ioutil.ReadFile(c.sigFile)
+		if err != nil {
+			t.Errorf("unable to read file %s", c.sigFile)
+			continue
+		}
+
+		server := initTestServer(
+			c.requestPath,
+			sigBytes,
+		)
+		defer server.Close()
+
+		awsc := &AwsClientImpl{
+			client: ec2metadata.New(unit.Session, &aws.Config{Endpoint: aws.String(server.URL + "/latest")}),
+		}
+
+		docBytes, err := awsc.getInstanceIdentityDocument()
+		if len(c.expectedErr) > 0 {
+			if err == nil {
+				t.Errorf("%s: Succeeded. Error expected: %v", id, err)
+			} else if err.Error() != c.expectedErr {
+				t.Errorf("%s: incorrect error message: %s VS %s",
+					id, err.Error(), c.expectedErr)
+			}
+			continue
+		} else if err != nil {
+			t.Fatalf("%s: Unexpected Error: %v", id, err)
+		}
+
+		doc := ec2metadata.EC2InstanceIdentityDocument{}
+		decode := json.NewDecoder(bytes.NewReader(docBytes)).Decode(&doc)
+		if decode != nil {
+			t.Fatalf("%s: Unexpected Error: %v", id, err)
+		}
+
+		if doc.InstanceType != c.expectedInstanceType {
+			t.Errorf("%s: Wrong Instance Type. Expected %s, Actual %s", id, c.expectedInstanceType, doc.InstanceType)
+		}
+
+		if doc.Region != c.expectedRegion {
+			t.Errorf("%s: Wrong Region. Expected %s, Actual %s", id, c.expectedRegion, doc.Region)
+		}
+	}
+}
+
+func TestAwsGetServiceIdentity(t *testing.T) {
+	testCases := map[string]struct {
+		sigFile                 string
+		requestPath             string
+		expectedErr             string
+		expectedServiceIdentity string
+	}{
+		"Good CredentialTypes": {
+			sigFile:                 "testdata/sig.pem",
+			requestPath:             "/latest/dynamic/instance-identity/pkcs7",
+			expectedErr:             "",
+			expectedServiceIdentity: "",
+		},
+	}
+
+	for id, c := range testCases {
+		sigBytes, err := ioutil.ReadFile(c.sigFile)
+		if err != nil {
+			t.Errorf("unable to read file %s", c.sigFile)
+			continue
+		}
+
+		server := initTestServer(
+			c.requestPath,
+			sigBytes,
+		)
+		defer server.Close()
+
+		awsc := &AwsClientImpl{
+			client: ec2metadata.New(unit.Session, &aws.Config{Endpoint: aws.String(server.URL + "/latest")}),
+		}
+
+		serviceIdentity, err := awsc.GetServiceIdentity()
+		if serviceIdentity != c.expectedServiceIdentity {
+			t.Errorf("%s: Wrong Service Identity. Expected %v, Actual %v", id,
+				string(c.expectedServiceIdentity), string(serviceIdentity))
+		}
+	}
+}
 
-	testcase := "Get Identity Document"
-	docBytes, err := na.getInstanceIdentityDocument()
+func TestGetGetAgentCredential(t *testing.T) {
+	testCases := map[string]struct {
```

Comment:
If there's just a single test case, you don't need to specify it as a map and iterate it via a `for` loop

Reply:
More test case can be added in the future.